packages/notifier/src/stored-topic.js

```diff
@@ -0,0 +1,54 @@
+import { assertAllDefined } from '@agoric/internal';
+import { makeMarshallToStorage } from '@agoric/internal/src/lib-chainStorage.js';
+import { M } from '@agoric/store';
+import { SubscriberShape } from './publish-kit.js';
+import { forEachPublicationRecord } from './storesub.js';
+
+export const TopicMetaShape = M.splitRecord(
+  {
+    topic: SubscriberShape,
+    storagePath: M.promise(/* string */),
+  },
+  { description: M.string() },
+);
+
+/**
+ * @template {object} T topic value
+ * @typedef {{
+ *   description?: string,
+ *   topic: Subscriber<T>,
+ *   storagePath: ERef<string>,
+ * }} TopicMeta
+ */
+
+export const TopicsRecordShape = M.recordOf(M.string(), TopicMetaShape);
+
+/**
+ * @typedef {{
+ *   [topicName: string]: TopicMeta<unknown>,
+ * }} TopicsRecord
+ */
+
+/**
+ * NB: caller must ensure that `publisher.finish()` or `publisher.fail()` is
+ * called when the publisher stores its final value.
+ * Otherwise this watch is retained and can't be GCed.
+ *
+ * @param {Subscriber<any>} topic
+ * @param {ERef<StorageNode>} storageNode
+ * @param {ERef<ReturnType<import('@endo/marshal').makeMarshal>>} marshaller
+ * @returns {void}
+ */
+export const pipeTopicToStorage = (topic, storageNode, marshaller) => {
```

Comment:
I particularly like `pipeTopicToStorage();`, rather than whatever implicit construction was going on in `makeStoredSubscriber`.  The semantics seem much clearer. I still don't completely understand it, but I suspect I'll be much more able to make use of it when I need to.

Reply:
One downside is that now there's the risk of forgetting to pipe (e.g. b00d09c1551369d9661571b918ea1423c434dce1) because the piping and describing the piping to consumers are separated.

After Vaults release I expect we'll improve these APIs to prevent that. But the goal of this PR is just to enable durable topics to unblock contract upgrade.